runtime/sema/gen.go

```diff
@@ -20,3 +20,4 @@ package sema
 
 //go:generate go run github.com/cheekybits/genny -pkg=sema -in=../common/orderedmap/orderedmap.go -out=ordered_map_string_type.go gen "KeyType=string ValueType=Type"
 //go:generate go run github.com/cheekybits/genny -pkg=sema -in=../common/orderedmap/orderedmap.go -out=ordered_map_string_member.go gen "KeyType=string ValueType=*Member"
+//go:generate go run github.com/cheekybits/genny -pkg=sema -in=../common/orderedmap/orderedmap.go -out=ordered_map_interfacetype_struct.go gen "KeyType=*InterfaceType ValueType=struct{}"
```

Comment:
I'm currently distinguishing between "primitive" ordered maps that are useful in multiple packages (e.g. string-to-string), and `sema`-specific maps (generated by this file, e.g. string-to-type).

We can't move the `sema`-specific types to `common`, because `common` is used by multiple other packages, and that would introduce a circular dependency (`common` is at the bottom of the dependency chain)

Reply:
I see. cool!